hawkular-inventory-parent/hawkular-inventory-model/src/main/java/org/hawkular/inventory/model/Resource.java

```diff
@@ -44,9 +47,11 @@
     private final String name;
 
     @JsonInclude(Include.NON_NULL)
+    @Field(store = Store.YES, analyze = Analyze.NO, indexNullAs = Field.DEFAULT_NULL_TOKEN)
```

Comment:
What's the options for "indexNullAs"? For information, "typeId" should never be null. "rootId" on the contrary can be null ; I don't know if it impacts anything here.

Reply:
it has little impact, basically is to decide which token will be used to stored null values.
By default, null values are skipped into the indexing engine and we want them at least for rootId.
Today we don't procect/validate per field (which is not bad as main users is agent), so we could remove this validation on typeId in a future iteration when we sync use cases of the agent.